README.md

```diff
@@ -40,189 +40,113 @@ When using GitLink, the user no longer has to specify symbol servers. The only r
+    GitLink.exe c:\source\catel -u "https://host/projects/catel/repos/catel/browse/{filename}?at={revision}&raw"
 
-    GitLink.exe -help
+The custom url will be used to fill the placeholders with the relative file path and the revision hash.
 
-## Logging to a file
+### Logging to a file
 
 When you need to log the information to a file, use the following command line:
 
     GitLink.exe c:\source\catel -u https://github.com/catel/catel -b develop -l GitLinkLog.log
 
-# Using GitLink in code
-
-GitLink is built with 2 usages in mind: command line and code reference. Though most people will use the command line version, it is possible to reference the executable and use the logic in code.
+### More options
 
-The command line implementation uses the same available API. 
+There are many more parameters you can use. Display the usage doc with the following command line:
 
-## Creating a context
+    GitLink.exe -h
 
-To link files to a Git project, a context must be created. The command line version does this by using the *ArgumentParser* class. It is also possible to create a context from scratch as shown in the example below:
+# How does it work
 
-	var context = new GitLink.Context(new ProviderManager());
-	context.SolutionDirectory = @"c:\source\catel";
-	context.TargetUrl = "https://github.com/catel/catel";
-	context.TargetBranch = "develop";
+The SrcSrv tool (Srcsrv.dll) enables a client to retrieve the exact version of the source files that were used to build an application. Because the source code for a module can change between versions and over the course of years, it is important to look at the source code as it existed when the version of the module in question was built.
 
-It is possible to create a context based on command line arguments:
+For more information, see the <a href="http://msdn.microsoft.com/en-us/library/windows/hardware/ff558791(v=vs.85).aspx" target="_blank">official documentation of SrcSrv</a>.
 
-	var context = ArgumentParser.Parse(@"c:\source\catel -u https://github.com/catel/catel -b develop");
+GitLink creates a source index file and updates the PDB file so it will retrieve the files from the Git host file handler.
+To do this, GitLink must be aware of the public URL from which the source files you compiled with can be retrieved.
+GitLink.exe reads your compiler-generated PDB, which already contains full paths to your local source files.
+It then searches for a git repo that contains those source files and looks up the commit that HEAD points to.
+It also searches your remotes for a URL pattern that it recognizes (e.g. https://github.com/name/repo).
+It combines the URL and the commit ID to create a unique URL for each source file of this exact version, and adds this information to your PDB.
 
-## Linking a context
+When you share your PDB alongside your DLL, your users who debug with Source Server support enabled will automatically be able to step into your source code. 
```

Comment:
Rename DLL to assembly, which is a better name for .net assemblies, and technically I can reference an exe and still use this approach